Pod/Classes/SEGAdobeIntegration.m

```diff
@@ -13,6 +13,121 @@
 #import "ADBMediaHeartbeat.h"
 
 
+@implementation SEGPlaybackDelegate
+
+- (instancetype)initWithPlayheadPosition:(long)playheadPosition
+{
+    if (self = [super init]) {
+        self.playheadPositionTime = CFAbsoluteTimeGetCurrent();
+        self.playheadPosition = playheadPosition;
+    }
+    return self;
+}
+
+/**
+ Adobe invokes this method once per second to resolve the current position of the videoplayhead. Unless paused, this method increments the value of playheadPosition by one every second by calling incrementPlayheadPosition
+ 
+ @return playheadPosition
+ */
+- (NSTimeInterval)getCurrentPlaybackTime
+{
+    if (self.isPaused) {
+        return self.playheadPosition;
+    }
+    return [self calculateCurrentPlayheadPosition];
+}
+
+/**
+ Called to stop the playheadPosition from incrementing.
+ Captures playheadPosition and playheadPositionTime so when
+ the playback resumes, the increment can continue from where the
+ player left off.
+ */
+- (void)pausePlayhead
+{
+    self.isPaused = true;
+    self.playheadPosition = [self calculateCurrentPlayheadPosition];
+    self.playheadPositionTime = CFAbsoluteTimeGetCurrent();
+}
+
+/**
+ Captures the playhead and current time when the video resumes.
+ */
+- (void)unPausePlayhead
+{
+    self.isPaused = false;
+    self.playheadPosition = [self calculateCurrentPlayheadPosition];
```

Comment:
@f2prateek You had self.playheadPosition = self.playheadPosition, but my thought here is after unpausing we should calculate the playhead position in case after unpausing the users place has changed. Thoughts?

Reply:
If the video has been paused at 12:35 and the position is 10, then when the user resumes at 12:45 the position is still 10 - it doesn't become 20 (10+12:45-12:35).

Reply:
Yeah that makes sense — updated!